### DISCUSSION TITLE
(feature request) Swim lane diagrams

### DISCUSSION THREAD

**Discussion by victorwestmann** [terrastruct/d2#236] [Feature Requests]
Hi everyone, first of all thank you so much for this fantastic piece of software!

I was just wondering if it is possible to add a new feature to d2. If we could add the possibility to do swim-lane diagrams for a future version of d2. We would be able to hav a top band with the title of the diagram, a subheader so we could add the last date of the diagram update and the name of the person who edited the diagram.

Also if it is possible to add options to change horizontal and vertical swim lanes with the same content.
Thank you for your time and for sharing high-quality work publicly!

Victor

**Comment by ejulio-ts** [comment]
@victorwestmann can you add some image examples so that we get a good idea around your request?

**Comment by victorwestmann** [comment]
Sure thing! This should be a great example of a swimlane diagram (this one seems to be done in Microsoft Visio).

![httpatomoreillycomsourcemspimages1696588](https://user-images.githubusercontent.com/582569/203694230-7ed5dbce-77fa-4972-9f7d-094ca9116395.png)

Overall what matters are 3 elements:
#1 The title of the diagram (i.e., "human resources Recruiting Swimlane Diagram")
#2 the subtitle of the diagram (i.e., "Phase", _but it could easily be "author" and "last date updated")_
#3 the lanes (i.e., "Recruiter", "HR Admin", "Hiring Manager", and "Candidate")

Please let me know if you have any questions or need anything else!
Thank you,
Victor

**Reply by jpike88** [reply]
I guess this is never happening

**Comment by asule90** [comment]
this would be great..i need this..
any progress on this idea?

i tried to achieve this utilizing grid..

![image](https://github.com/terrastruct/d2/assets/10544636/1abc1ffa-d483-4496-a2c6-a63936089446)

**Reply by muteboy** [reply]
I got a similar result with borders and grid-gap. The problem is that if a shape in one lane container links to a shape in another lane container, the target shape is not forced below the source shape.

https://play.d2lang.com/?script=TIxLDoIwEIb3PcV_AYjArguvYvoY2iaFGithQXp3M4NEN5Nv_lfd05LNSgiv5DUOBaHOlbwta9WYLiWYp8ZNARwfzihg0N1hBS3jo2d77Ekkx5I_UfOHd0wVNZYte1iCpVx26bfv8ngte-4Sn1n--Tc_9O6vMWkcgc3IJzXV1CcAAP__&sketch=0&

**Reply by brussee** [reply]
@muteboy Possible workaround:
```
    b -- c: {
      style.stroke: transparent
    }
```
https://play.d2lang.com/?script=TI5NbsQgDIX3nOJdIKgzs2PRq1T8OAkqgQiIoiri7pVJUWdjPb_Pz3Y5_RZ0JCzZO4VLoKvJpnBssSi8hrPoXeFDADz-uEcBjekTpkvD8ksyfkrqlmXLDTrBjhhQ6k8gWWpO36RQs45l15li7bzdccUbUFdfUNZ0BAdDMBTS2W-2v2-eY63je8Rl7v38_9JD2rfES-FaGK5cfBNN_AYAAP__&sketch=0&

**Reply by muteboy** [reply]
Good idea. Worth trying until the real thing comes. Thanks!

**Comment by wiztensai** [comment]
up up uppp
